### ISSUE TITLE
Issue with 'resettable' attribute/directive

### ISSUE THREAD

**Issue by pierznj** [facultymatt/angular-unsavedChanges#19]
Thanks for this implementation - very useful. However, the 'resettable' attribute can cause issues. It seems fine with a simple `ng-model` binding, but if there is some function called on `ng-change`, the value of the object will not have changed. For example I have:

```
<select id="owner" 
    class="form-control" 
    ng-model='owner' 
    resettable 
    required 
    ng-change="selectOwner()"  
    ng-options="user.friendlyName for user in owners" />
```

Now if after selecting some new value I examine the object `$scope.owner` inside the `ng-change` function it will not be the new value but the old one. I've not been able to work out yet from the directive code why this should be so.

**Comment by facultymatt** [comment]
If you provide a plunkr that reproduces this error I will be able to address it much sooner...

**Comment by facultymatt** [comment]
Fixed in v0.2.3-alpha.2
